dhcpv6/nclient6/conn_linux.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 the u-root Authors and Andrea Barberio. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package nclient6
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+// NewIPv6UDPConn returns a UDP connection bound to both the interface and port
+// given based on a IPv4 DGRAM socket. The UDP connection allows broadcasting.
+func NewIPv6UDPConn(iface string, port int) (net.PacketConn, error) {
+	fd, err := unix.Socket(unix.AF_INET6, unix.SOCK_DGRAM, unix.IPPROTO_UDP)
```

Comment:
question (not a request to change): do we need an explicit socket at all with v6? We are guaranteed to always have a link-local address, so net.ListenUDP should be enough.

Partial answer to my own question is that with unix.Socket we can use SO_REUSEADDR and BindToDevice, which is only necessary in case of misconfigurations (e.g. two NICs with the same MAC address, but I'd call it extremely rare) or interface aliases

Reply:
good point -- BindToDevice is taken care off by the Zone config, and we could add reuseaddr to the UDPConn we get, too, if needed.